NOMIS/nomis.go

```diff
@@ -301,3 +307,25 @@ func downloadFile() {
 
 	fmt.Printf("Downloaded a file %s with size %d", fileName, size)
 }
+
```

Comment:
General observation: This is reasonably complex string parsing logic so are there any unit tests to cover functionality?

I'm aware this might be a lot of work to implement but I feel it's reasonable to require to tests to accompany something of this nature. If nothing else it gives you confidence the code behaves as expected and  will behave correctly given different inputs.